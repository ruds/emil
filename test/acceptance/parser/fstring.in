f"it's $foo in this $bar!";
f"Therefore, 'a' is ${let
  val a = 1
in
  a
end}";
f"Here's a record: ${{foo=1, bar=2}}";
