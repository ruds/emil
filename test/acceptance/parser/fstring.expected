0001 (ExprTopDecl
         :expr
         (FstringLiteralExpr
             :segments
             (
                 "6974277320"
                 "20696E207468697320"
                 "21"
             )
             :substitutions
             (
                 (IdentifierExpr
                     :qualifiers
                     ()
                     :identifier
                     "666F6F"
                     :is_op
                     false
                     :is_prefix_op
                     false)
                 (IdentifierExpr
                     :qualifiers
                     ()
                     :identifier
                     "626172"
                     :is_op
                     false
                     :is_prefix_op
                     false)
             )))
0002 (ExprTopDecl
         :expr
         (FstringLiteralExpr
             :segments
             (
                 "5468657265666F72652C2027612720697320"
                 ""
             )
             :substitutions
             ((LetExpr
                 :decls
                 ((ValDecl
                     :bindings
                     (((IdentifierPattern
                         :qualifiers
                         ()
                         :identifier
                         "61"
                         :is_op
                         false
                         :is_prefix_op
                         false)
                       . (BigintLiteralExpr
                             :val
                             1)))))
                 :exprs
                 ((IdentifierExpr
                     :qualifiers
                     ()
                     :identifier
                     "61"
                     :is_op
                     false
                     :is_prefix_op
                     false))))))
0007 (ExprTopDecl
         :expr
         (FstringLiteralExpr
             :segments
             (
                 "4865726527732061207265636F72643A20"
                 ""
             )
             :substitutions
             ((RecordExpr
                 :rows
                 (
                     (RecRowExpr
                         :label
                         "666F6F"
                         :val
                         (BigintLiteralExpr
                             :val
                             1))
                     (RecRowExpr
                         :label
                         "626172"
                         :val
                         (BigintLiteralExpr
                             :val
                             2))
                 )))))
0008 (EndOfFileTopDecl)
